Maple2.Model/Game/Shop/BeautyShopCost.cs

```diff
@@ -5,28 +5,26 @@
 namespace Maple2.Model.Game.Shop;
 
 public class BeautyShopCost : IByteSerializable {
-    public static BeautyShopCost Zero = new BeautyShopCost(ShopCurrencyType.Meso, 0);
+    public static BeautyShopCost Zero = new BeautyShopCost(ShopCurrencyType.Meso, 0, 0);
 
-    public readonly ShopCurrencyType Type;
-    public readonly int ItemId;
-    public readonly int Amount;
+    public ShopCurrencyType Type { get; init; }
+    public int ItemId { get; init; }
+    public int Amount { get; init; }
+    public string ItemIdIconCode { get; init; }
 
-    public BeautyShopCost(ShopCurrencyType type, int amount) {
+    public BeautyShopCost(ShopCurrencyType type, int itemId, int amount) {
         Type = type;
-        ItemId = 0;
-        Amount = amount;
-    }
-
-    public BeautyShopCost(int itemId, int amount) {
-        Type = ShopCurrencyType.Item;
         ItemId = itemId;
         Amount = amount;
+        
+        // TODO: This is a hack to get the icon code for the item ID. This should be replaced with a proper lookup.
+        ItemIdIconCode = ItemId is >= 20500001 and < 20600001 ? "20500001" : string.Empty;
```

Comment:
are icon codes just the id as a string or is this a random value for now?

Reply:
It _sometimes_ is the string of an id. The codes are specific ids in order to display the proper currency icon in shops. I'm not really sure where they're truly defined. 
Not every item can be shown as a currency icon in shops - only selects. These are all the valid ids, provided by @gwagwaflamingo 

> 30001490
20300165
30001393
30001165
30001182
30001190
30000898
30001304
30001605
30001477
30001437
30000855
20300123
30000448
30000895
30001153
30001344
30000897
30000913
30001215
30001203
30001091
30001196
30001217
40100027
20301082
30000466
30001101
20510001
20510001
20500001
30000861
30001342
30001463
20300154
30001049
30001048
30001047
30001050
30001046
30000925
30001216
30001394
40100029
30001135
30001277
30001105
30001124
30001281
30001280
30001279
30001376
30001369
30001154
30000782
30000554
22204001
22202001
22201001
30000447
30000419
30001505
20300246
30000867
30001444

Reply:
For the record, do you know how these values are supposed to be determined?

Reply:
No idea, tbh.